### PR TITLE
auto download latest rust-analyzer

### PR DESCRIPTION
**Pull request by prabirshrestha**

### CODE REVIEW

**Comment by lnicola** [comment]
Please download the compressed binaries instead. We want to stop uploading the uncompressed ones, and some platforms don't even have those.

**Comment by prabirshrestha** [comment]
@lnicola any reason why rust-analyzer is not using `.tar.gz` instead? Most linux ships with tar and curl. Windows ships it too. https://docs.microsoft.com/en-us/virtualization/community/team-blog/2017/20171219-tar-and-curl-come-to-windows

**Comment by lnicola** [comment]
If you can unpack .tar.gz, you can unpack .gz, since they're different formats, layered on top of each other. We don't use TAR because it's used to put multiple files in an archive and rust-analyzer is a single file.

From what I've seen, ZIP seems more used, so I kinda wish we've gone with that. unzip isn't installed by default on Linux systems, though.

File: installer/install-rust-analyzer.sh [hunk 1]
```diff
@@ -13,7 +13,12 @@ darwin)
...
 esac
 
-version="2021-01-18"
+if [ ! which grep &> /dev/null ] || [ ! which cut &> /dev/null ]; then
+  version="2021-02-15"
+else
+  version=$(curl -sL https://api.github.com/repos/rust-analyzer/rust-analyzer/releases/latest | grep '"tag_name":' | cut -d'"' -f4)
+fi
+
 curl -L -o rust-analyzer-$platform "https://github.com/rust-analyzer/rust-analyzer/releases/download/$version/rust-analyzer-$platform"
```
Comment (heavenshell) [inline installer/install-rust-analyzer.sh:22]: [IMO]
`latest` can download latest release.

e.g.
`curl -L -o "rust-analyzer-$platform" "https://github.com/rust-analyzer/rust-analyzer/releases/latest/download/rust-analyzer-$platform"`

Is this enough?

**Comment by wordijp** [comment]
In the future, I think we can compare the latest version check with the commit hash of `$ rust-analyzer --version` and `grep'"target_commitish":' in JSON.
So I want to keep the code added in this PR related so that it doesn't seem redundant.

File: installer/install-rust-analyzer.sh (hunk 1 above)
Comment (prabirshrestha) [reply installer/install-rust-analyzer.sh]: fixed

**Comment by prabirshrestha** [comment]
PR should be ready for second look.

I have filed an issue on rust-analyzer to support .tar.gz https://github.com/rust-analyzer/rust-analyzer/issues/7747

**Comment by prabirshrestha** [comment]
For windows we would need to have [gunzip.exe ](http://www2.cs.uidaho.edu/~jeffery/win32/). I have tried extracting it and seems to work. @mattn  Could you upload `gunzip.exe` to https://github.com/mattn/vim-lsp-settings/releases/tag/v0.0.1 so I can download it in the script if you are ok with it. Another option I thought was to upload 7zip but it contains a bunch of other dlls.

I still think it would be better for rust-analyzer to support `.tar.gz` or `.zip` for windows.

For unix systems I can try to check if `gunzip` exists and use that else directly download the executable.

**Comment by mattn** [comment]
Let's use latest executable for Rust.

**Pull request merged**